### DISCUSSION TITLE
.env folde/file is not even getting created

### DISCUSSION THREAD

**Discussion by AmirAlasady** [cookiecutter/cookiecutter-django#5550] [Q&A]
when i create a project its not being created with env folder or dir 
as im just having the 
    import environ
ModuleNotFoundError: No module named 'environ'
error. 
when i try to run the project with 
python3 -m manage runserver

**Comment by foarsitter** [comment]
Programming is like a puzzle, you need to find the correct piece to be able to continue. Unlike a jigsaw puzzle you have a search function. Searching with the term `No module named 'environ'` brings me an answer from 2016: https://github.com/cookiecutter/cookiecutter-django/issues/965#issuecomment-269779035  

But we live in 2024 and we have AI, so I asked ChatGPT: 

> I'm running into the following error with django: No module named 'environ'

The error "No module named 'environ'" indicates that Python cannot find the `environ` module, which is typically part of the `django-environ` package. This package is used for managing environment variables in Django projects.

Here’s how to resolve the issue:

1. **Install django-environ**: If you haven’t installed it yet, you can do so using pip. Run the following command in your terminal:

   ```bash
   pip install django-environ
   ```

2. **Check your virtual environment**: If you're using a virtual environment, make sure it is activated before you run the install command. You can activate it by navigating to your project directory and running:

   - On Windows:
     ```bash
     .\venv\Scripts\activate
     ```
   - On macOS/Linux:
     ```bash
     source venv/bin/activate
     ```

3. **Verify the installation**: After installation, you can verify that `django-environ` is installed by running:

   ```bash
   pip show django-environ
   ```

4. **Update your imports**: Ensure that your import statement in your Django settings or any other module is correct:

   ```python
   import environ
   ```

5. **Check for typos**: Sometimes, the error might be caused by a simple typo in the import statement.

After performing these steps, try running your Django project again. If you continue to encounter issues, feel free to provide more details about your setup!

If this solve the problem, please, close this issue. If not, let me know so I can help you further.

**Comment by AmirAlasady** [comment]
I've tried everything and gave it a really long time. I've been trying and
troubleshooting that for almost 1 month from now
, asking ai models , yt tutorials , books , asking my friends , trying
different machines etc ...
but all seems to lead to nothing .
The documentation does not seem to be well user friendly
something is really out of the ordinary .
and a lot of things are not clear from more than one version and all are
different to the problem of what comes 1st,  problems the end or the
project and every time i need to install everything manually. and ain't the
end nothing works trust me when i say that i've literally tried
everything . I DID try everything and I'm just so frustrated . and wasted
my time on the project . so pls if u got any idea i really really need ur
help .

**Comment by foarsitter** [comment]
Two things:

The command should be: 
python3 manage.py runserver

The -m argument means invoking a module called 'manage'. Be sure you are in the correct virtual environment. 

Second: follow the issue template so we know what kind of OS and python version you are on.
